{"uid":"61f0ee3effd108a5","name":"test_case[dic0]","fullName":"testCase.test_case.TestCase#test_case","historyId":"1eb211e27567e2ea623bd553ee5062cc","time":{"start":1683368510251,"stop":1683368512992,"duration":2741},"status":"failed","statusMessage":"AssertionError: 预期结果与实际结果不符，预期结果：{'errorCode':'0','username':'liusha'},实际结果：0\nassert '0' == \"{'errorCode'...me':'liusha'}\"\n  - {'errorCode':'0','username':'liusha'}\n  + 0","statusTrace":"self = <testCase.test_case.TestCase object at 0x10c2af250>\ndic = {'caseid': '', 'expect': \"{'errorCode':'0','username':'liusha'}\", 'header': '{}', 'id': 1.0, ...}\n\n    @pytest.mark.parametrize(\"dic\", test_data)\n    def test_case(self, dic):\n        logger.debug(f\"====>{dic}\")\n        # 判断是否有依赖,执行依赖接口替换依赖字段\n        ps = PreSolve(test_data)\n        dic[\"header\"], dic[\"value\"] = ps.presolve(dic)\n        #实例化\n        ch = ConfigHttp(dic)\n        #调用实例方法\n        res = ch.run()\n        # 实例化\n        pa = PublicAssert(dic, res)\n        # 调用实例方法断言\n        pa.public_assert()\n        # 将响应的json数据转化成字典\n        res_dict = res.json()\n    \n        # 坑一：用if断言需在失败情况下抛出assertion\n        # if str(res_dict[\"errorCode\"]) == str(dic[\"expect\"]):\n        #     print(\"用例执行成功\")\n        # else:\n        #     print(\"用例执行失败\")\n        #     raise AssertionError(\"预期结果与实际结果不符\")\n    \n        # 坑二：try-except不要加在self.assertEqual前后，否则unittest捕获不到AssertionError,就默认为用例通过\n        # try:\n        #     assert str(res_dict[\"errorCode\"]) ==str(dic[\"expect\"]), f\"预期结果与实际结果不符，预期结果：{dic['except']},实际结果：{res_dict['errorCode']}\"\n        # except Exception as msg:\n        #     print(f\"报错信息：{msg}\")\n        #     raise\n    \n        # 2.1.3断言返回的实际结果，断言用例执行成功/失败\n>       assert str(res_dict[\"errorCode\"]) == str(dic[\"expect\"]), f\"预期结果与实际结果不符，预期结果：{dic['expect']},实际结果：{res_dict['errorCode']}\"\nE       AssertionError: 预期结果与实际结果不符，预期结果：{'errorCode':'0','username':'liusha'},实际结果：0\nE       assert '0' == \"{'errorCode'...me':'liusha'}\"\nE         - {'errorCode':'0','username':'liusha'}\nE         + 0\n\ntestCase/test_case.py:60: AssertionError","flaky":false,"newFailed":false,"newBroken":false,"newPassed":false,"retriesCount":0,"retriesStatusChange":false,"beforeStages":[],"testStage":{"status":"failed","statusMessage":"AssertionError: 预期结果与实际结果不符，预期结果：{'errorCode':'0','username':'liusha'},实际结果：0\nassert '0' == \"{'errorCode'...me':'liusha'}\"\n  - {'errorCode':'0','username':'liusha'}\n  + 0","statusTrace":"self = <testCase.test_case.TestCase object at 0x10c2af250>\ndic = {'caseid': '', 'expect': \"{'errorCode':'0','username':'liusha'}\", 'header': '{}', 'id': 1.0, ...}\n\n    @pytest.mark.parametrize(\"dic\", test_data)\n    def test_case(self, dic):\n        logger.debug(f\"====>{dic}\")\n        # 判断是否有依赖,执行依赖接口替换依赖字段\n        ps = PreSolve(test_data)\n        dic[\"header\"], dic[\"value\"] = ps.presolve(dic)\n        #实例化\n        ch = ConfigHttp(dic)\n        #调用实例方法\n        res = ch.run()\n        # 实例化\n        pa = PublicAssert(dic, res)\n        # 调用实例方法断言\n        pa.public_assert()\n        # 将响应的json数据转化成字典\n        res_dict = res.json()\n    \n        # 坑一：用if断言需在失败情况下抛出assertion\n        # if str(res_dict[\"errorCode\"]) == str(dic[\"expect\"]):\n        #     print(\"用例执行成功\")\n        # else:\n        #     print(\"用例执行失败\")\n        #     raise AssertionError(\"预期结果与实际结果不符\")\n    \n        # 坑二：try-except不要加在self.assertEqual前后，否则unittest捕获不到AssertionError,就默认为用例通过\n        # try:\n        #     assert str(res_dict[\"errorCode\"]) ==str(dic[\"expect\"]), f\"预期结果与实际结果不符，预期结果：{dic['except']},实际结果：{res_dict['errorCode']}\"\n        # except Exception as msg:\n        #     print(f\"报错信息：{msg}\")\n        #     raise\n    \n        # 2.1.3断言返回的实际结果，断言用例执行成功/失败\n>       assert str(res_dict[\"errorCode\"]) == str(dic[\"expect\"]), f\"预期结果与实际结果不符，预期结果：{dic['expect']},实际结果：{res_dict['errorCode']}\"\nE       AssertionError: 预期结果与实际结果不符，预期结果：{'errorCode':'0','username':'liusha'},实际结果：0\nE       assert '0' == \"{'errorCode'...me':'liusha'}\"\nE         - {'errorCode':'0','username':'liusha'}\nE         + 0\n\ntestCase/test_case.py:60: AssertionError","steps":[],"attachments":[{"uid":"cc92ece388423f13","name":"log","source":"cc92ece388423f13.txt","type":"text/plain","size":378}],"parameters":[],"stepsCount":0,"attachmentsCount":1,"shouldDisplayMessage":true,"hasContent":true},"afterStages":[],"labels":[{"name":"parentSuite","value":"testCase"},{"name":"suite","value":"test_case"},{"name":"subSuite","value":"TestCase"},{"name":"host","value":"liushadeMBP"},{"name":"thread","value":"9783-MainThread"},{"name":"framework","value":"pytest"},{"name":"language","value":"cpython3"},{"name":"package","value":"testCase.test_case"},{"name":"resultFormat","value":"allure2"}],"parameters":[{"name":"dic","value":"{'id': 1.0, 'interfaceUrl': 'https://www.wanandroid.com/user/login', 'name': 'login', 'method': 'post', 'value': \"{'username':'liusha','password':'123456'}\", 'header': '{}', 'rely': 'n', 'caseid': '', 'expect': \"{'errorCode':'0','username':'liusha'}\"}"}],"links":[],"hidden":false,"retry":false,"extra":{"severity":"normal","retries":[],"categories":[{"name":"Product defects","matchedStatuses":[],"flaky":false}],"tags":[]},"source":"61f0ee3effd108a5.json","parameterValues":["{'id': 1.0, 'interfaceUrl': 'https://www.wanandroid.com/user/login', 'name': 'login', 'method': 'post', 'value': \"{'username':'liusha','password':'123456'}\", 'header': '{}', 'rely': 'n', 'caseid': '', 'expect': \"{'errorCode':'0','username':'liusha'}\"}"]}